app/src/main/kotlin/com/exallium/todoapp/notedetail/NoteDetailPresenter.kt

```diff
@@ -0,0 +1,38 @@
+package com.exallium.todoapp.notedetail
+
+import com.exallium.todoapp.entities.Note
+import com.exallium.todoapp.mvp.BasePresenter
+import rx.Single
+import rx.SingleSubscriber
+import timber.log.Timber
+
+/**
+ * Presenter for Note Detail Screen
+ */
+class NoteDetailPresenter(val view: NoteDetailView, val model: NoteDetailModel) : BasePresenter<NoteDetailView>(view) {
+    lateinit var note: Note
+
+    override fun onViewCreated() {
+        if (view.isCached()) {
+            return
+        }
+        lookupNoteDetail(view.getNoteDetailId())
+        view.setNoteData(note)
+    }
+
+    fun lookupNoteDetail(noteId: String) {
+        var noteSingle: Single<Note> = model.getNoteById(noteId)
```

Comment:
I would not bother breaking this down like this:

```
compositeSubscription.add(model.getNoteById(noteId).subscribe(/* ... */))
```

At the most, I'd break subscription into it's own variable, but I don't think noteSingle necessarily makes this more readable.